<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.2.269">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Zane">
<meta name="dcterms.date" content="2023-02-06">

<title>Partial pooling test</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1.6em;
  vertical-align: middle;
}
</style>


<script src="Partial-Pooling-Test_files/libs/clipboard/clipboard.min.js"></script>
<script src="Partial-Pooling-Test_files/libs/quarto-html/quarto.js"></script>
<script src="Partial-Pooling-Test_files/libs/quarto-html/popper.min.js"></script>
<script src="Partial-Pooling-Test_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="Partial-Pooling-Test_files/libs/quarto-html/anchor.min.js"></script>
<link href="Partial-Pooling-Test_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="Partial-Pooling-Test_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="Partial-Pooling-Test_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="Partial-Pooling-Test_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="Partial-Pooling-Test_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">

  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">Partial pooling test</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Zane </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 6, 2023</p>
    </div>
  </div>
  
    
  </div>
  

</header>

<p>Based on conversations with Amanda and Andreas, we decided to switch from trying to make a CI out of the individual fits to fitting a multilevel model using a Bayesian partial pooling approach. This script is my first test of this idea.</p>
<section id="introduction" class="level1">
<h1>Introduction</h1>
<p>One of the main concepts for this paper was to fit linear regressions on titer increase vs.&nbsp;antigenic distance using the UGAFluVac data. This linear model acts like a summary of the information contained in the antigenic distance data, and can be used to quantify both the strength and breadth of the response. Going forward we adopt this terminology:</p>
<ul>
<li><strong>Strength</strong> of the response: quantifies how much of an immune response is generated to the homologous strain.</li>
<li><strong>Breadth</strong> of the response: quantifies how much of an immune response is generated to antigenically distinct strains.</li>
<li><strong>Overall magnitude of the response</strong>: combines strength and breadth together into an idea of how much of an immune response was induced overall.</li>
</ul>
<p>The basic idea of our framework for quantifying these things is that the AUC of the linear regression line estimates the overall magnitude of the response, and depends on the slope and intercept of the line. The slope captures information about the breadth of the response while the intercept captures information about the strength of the response. (See the <code>trapezoids.md</code> doc for a few brief thoughts I had about this.)</p>
<p>In the first round of tests, we used a simple linear regression model (a “complete pooling” model in bayesian terminology) as the estimator. (<a href="#fig-thesis">Figure&nbsp;1</a>).</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-thesis" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="Amanda-Email-Chain/0-Thesis-4p2.PNG" class="img-fluid figure-img" width="507"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;1: From Amanda’s thesis: The titer increase to the H1N1 and H3N2 virus panels for the 2017 season of all individuals who received SD vaccination. The columns are separated by vaccine strain. The linear regression of the titerincrease with 95% confidence intervals is shown for each distance method. The distances were normalized by season. The raw data points had jitter applied with +/- 0.4 in the y-axis. Raw data points that fell outside of the y-axis bounds are not shown.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>However, we then realized that there was a lot of variation in trajectories between individuals. <a href="#fig-indivLines">Figure&nbsp;2</a> shows the regression line for each individual. After seeing this, one of our main concerns was that there appears to be a significant amount of heterogeneity in the data. Therefore, it would be better to estimate the variability in the fitted regression line from this “no pooling” model rather than from the complete pooling model.</p>
<div class="cell">
<div class="cell-output-display">
<div id="fig-indivLines" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="Amanda-Email-Chain/1-Individual-Lines-Pre-Post-alt.png" class="img-fluid figure-img" width="326"></p>
<p></p><figcaption class="figure-caption">Figure&nbsp;2: From Amanda’s email: I’ve added the individual linear regression lines in the background. However, with the two colors/linetypes it’s very difficult to distinguish the main two lines for the groupings. This is an image that would go into Figure 1. Also it’s for MI/15 instead of CA/09 so there about 200 less individuals (400 less lines considering pre/post titer) present on these plots compared to what would be later for CA/09. Also, with the change in the HAI virus panel the domains of the linear regressions are different. Although it provides more transparency in that aspect it does add to the confusion.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p>From this figure, we can see that there are significant amount of people who had a much smaller panel that didn’t include the most distant viruses. Therefore, points with moderate antigenic distances have much more leverage and systematically drive their slopes down. <strong>We still need to investigate whether</strong> <strong>these points have similarly low values for patients with the full panel.</strong> There are two explanations here from my perspective.</p>
<ol type="1">
<li>These people with steeper slopes have systematic differences that make their responses less broad.</li>
<li>Everyone has a dip in the middle, but for people with the entire range of historical viruses in their panel, these points have much lower leverage and the regression line smooths over this bump in the trajectory.</li>
</ol>
<p>Anyways, we then ran into trouble trying to get some kind of average estimate (marginalized over individuals) from the no pooling estimate, but this is quite hard. <a href="#fig-bounds">Figure&nbsp;3</a> shows Amanda’s attempts at combining the average slope and intercept into boundaries. My proposal for this method was to take the empirical quantiles of the regression predictions over every x-value, but this got us into the weeds of the best way to compute this interval.</p>
<div id="fig-bounds" class="cell quarto-layout-panel">
<figure class="figure">
<div class="quarto-layout-row quarto-layout-valign-top">
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 47.1%;justify-content: center;">
<div id="fig-bounds-1" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="Amanda-Email-Chain/2-Bounding-Lines.png" class="img-fluid figure-img" data-ref-parent="fig-bounds" width="336"></p>
<p></p><figcaption class="figure-caption">(a) I’ll include the bounding lines for the titer = slope_upperSD * distance + intercept_upperSD (purple line) and opposite for lower bound (green line) since they include the other combinations (slope_lwr + intercept_upr (pink line).</figcaption><p></p>
</figure>
</div>
</div>
<div class="cell-output-display quarto-layout-cell quarto-layout-cell-subref" style="flex-basis: 52.9%;justify-content: center;">
<div id="fig-bounds-2" class="quarto-figure quarto-figure-center anchored">
<figure class="figure">
<p><img src="Amanda-Email-Chain/3-Individual-Lines-biggest-bounds.png" class="img-fluid figure-img" data-ref-parent="fig-bounds" width="377"></p>
<p></p><figcaption class="figure-caption">(b) The linear regression and the averaged Linear regression do not match. If the ranges of the data were the same it wouldn’t be a problem but the data that does not have as large of a range have a steeper slope and are pulling the averaged linear regression slope results to be steeper. This isn’t matched in the linear regression with all of the data since it doesn’t extrapolate for those points.</figcaption><p></p>
</figure>
</div>
</div>
</div>
<p></p><figcaption class="figure-caption">Figure&nbsp;3: From emails.</figcaption><p></p>
</figure>
</div>
<p>So Andreas and I both agreed that one potential solution to this CI issue would be <strong>partial pooling</strong>: using a multilevel model to allow the individual slopes to borrow from the overall mean if they are low precision or extremely different, but allowing for much more individual variation. We could use a frequentist mixed-effects model (e.g.&nbsp;through <code>lme4</code>) for this, but we also decided it would be easier to switch to a Bayesian hierarchical model at this time.</p>
</section>
<section id="the-model-math-part" class="level1">
<h1>The model (math part)</h1>
<p>For this example, we’ll only fit a model to the 2017 season and the H1N1-Michigan-2015 strain as a proof-of-concept. Then we can incorporate season and strain effects as well. Here’s the model I’ll attempt to fit. <span class="math display">\[
\begin{align*}
\text{HAI titer}_{i, t} &amp;\sim \text{Normal}\left(\mu_{i, t}, \sigma_{t} \right) \\
\mu_{i, t} &amp;= \alpha_{i, t} + \beta_{i, t} \left( \text{Ag distance} \right)_{i, t} \\
\alpha_{i, t} &amp;\sim \text{Normal}\left(5, 5\right) \\
\beta_{i, t} &amp;\sim \text{Normal}\left(0, 1\right) \\
\sigma_{i, t} &amp;\sim \text{Exponential}\left(0.25\right)
\end{align*}
\]</span></p>
<p><span class="math display">\[\begin{equation}
\begin{aligned}
\text{HAI titer}_{i, t} &amp;\sim \text{Normal}\left(\mu_{i, t}, \sigma_{t} \right) \\
\mu_{i, t} &amp;= \alpha_{i, t} + \beta_{i, t} \left( \text{Ag distance} \right)_{i, t} \\
\alpha_{i, t} &amp;\sim \text{Normal}\left(5, 5\right) \\
\beta_{i, t} &amp;\sim \text{Normal}\left(0, 1\right) \\
\sigma_{i, t} &amp;\sim \text{Exponential}\left(0.25\right)
\end{aligned}
\end{equation}\]</span></p>
<p>In this model, <span class="math inline">\(i\)</span> indexes the individuals, and <span class="math inline">\(t\)</span> is an index variable denoting whether the measurement is pre- or post-vaccination. Note that this naive model does not implement a correlation between the pre- and post-vaccination parameters for a given individual. Basically this fits completely separate models for the pre and post vaccination titers, but they are bundled together into one neat model formula.</p>
</section>
<section id="model-fitting-computational-part" class="level1">
<h1>Model fitting (computational part)</h1>
</section>
<section id="results" class="level1">
<h1>Results</h1>
<!-- END OF FILE -->
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    target: function(trigger) {
      return trigger.previousElementSibling;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn) {
    const config = {
      allowHTML: true,
      content: contentFn,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start'
    };
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->



</body></html>